docs/migrations/migrate_renderer.md

```diff
@@ -0,0 +1,228 @@
+    endDate: "2019-06-10T00:00:00+08:00"
+  }
+};
+```
+
+## Creating the template
+
+1. Replace the content of the React component in `src/templates/customTemplate/customTemplate.tsx` by the content of the [2018 Geekout](https://github.com/OpenCerts/legacy-templates/blob/master/src/components/CertificateTemplates/tlds/sg/gov/tech/2018-Geekout/certificate.js). You will see some errors that we will fix in the following steps.
+1. In `src/templates/customTemplate/customTemplate.tsx`:
+
+   1. Update the component declaration
+
+   ```tsx
+   const Template = ({ certificate }) => {
+   ```
+
+   by
+
+   ```tsx
+   export const CustomTemplate: FunctionComponent<TemplateProps<CustomTemplateCertificate>> = ({ document }) => {
+   ```
+
+   This change will help us to make sure our template is correctly typed.
+
+   > The component name does not matter much. However if we change the name `CustomTemplate` we will need to change the whole configuration. To make it simple we will stick to the name that was available in the renderer when we initialized it.
+
+   1. Remove the first line `import PropTypes from "prop-types";` which is useless because of Typescript.
+   1. Remove the 4 last lines which are useless because we don't need prop-types anymore, and the component has already been exported at a previous step.
+
+   ```jsx
+   Template.propTypes = {
+     certificate: PropTypes.object.isRequired
+   };
+   export default Template;
+   ```
+
+   1. Add the following imports
+
+   ```tsx
+   import React, { FunctionComponent } from "react";
+   import { TemplateProps } from "@govtechsg/decentralized-renderer-react-components";
+   import { CustomTemplateCertificate } from "../samples";
+   ```
+
+   `TemplateProps`, `CustomTemplateCertificate` and `FunctionComponent` are needed for correctly typing the template.
+
+   1. Add the following imports
+
+1. The template requires the use of `lodash` and `date-fns`. Let's add the libraries into the repository:
+
+```bash
+npm install lodash date-fns
+```
+
+There will be remaining error at `import css from "./styles.scss";` and when using the css variable, that we will cover in the following steps.
+
+## Creating the style
+
+The legacy renderer is configured to work with [SASS](https://sass-lang.com/) while the renderer what we used is configured to work with [Emotion](https://github.com/emotion-js/emotion). Both options are valid, however we will need either to convert the code to work with Emotion, or to configure the template to work with SASS. During the following step, we will focus on the latter option.
+
+1. Copy the [style file](https://github.com/OpenCerts/legacy-templates/blob/master/src/components/CertificateTemplates/tlds/sg/gov/tech/2018-Geekout/styles.scss) next to `src/templates/customTemplate/customTemplate.tsx`. Keep the filename untouched (`styles.css`).
+1. Add the following lines into `src/@types/declaration.d.ts`:
+
+```ts
+declare module "*.scss" {
+  interface IClassNames {
```

Comment:
Replace IClassNames by ClassNames to avoid this error:
<img width="770" alt="Screenshot 2020-10-06 at 11 37 51 AM" src="https://user-images.githubusercontent.com/38589870/95164556-c38e1400-07dc-11eb-9cb4-1a3668d9ff0a.png">

Reply:
done